opennms-provision/opennms-provisiond/src/main/java/org/opennms/netmgt/provision/service/operations/SaveOrUpdateOperation.java

```diff
@@ -124,6 +122,10 @@ public void foundInterface(InetAddress addr, Object descr, final PrimaryType pri
         //FIXME: verify this doesn't conflict with constructor.  The constructor already adds this
         //interface to the node.
         m_node.addIpInterface(m_currentInterface);
+
+        if (System.getProperty("org.opennms.provisiond.reverseResolveRequisitionIpInterfaceHostnames", "true").equalsIgnoreCase("true")) {
+            dnsLookups.add(CompletableFuture.supplyAsync(() -> getProvisionService().getHostnameResolver().getHostname(addr, m_node.getLocation().getLocationName())).thenAccept(s -> m_node.getInterfaceWithAddress(addr).setIpHostName(s)));
```

Comment:
Should we use some kind of thread pool here to limit this somehow?

Reply:
Yes, I will work something in.